src/Injector.php

```diff
@@ -171,22 +181,9 @@ protected function createInstance(string $name, array $params)
             ));
         }
 
-        $definition = $this->definition->getClassDefinition($class);
         $callParameters = $this->resolveParameters($name, $params);
 
-        // Hack to avoid Reflection in most common use cases
-        switch (count($callParameters)) {
-            case 0:
-                return new $class();
-            case 1:
-                return new $class($callParameters[0]);
-            case 2:
-                return new $class($callParameters[0], $callParameters[1]);
-            case 3:
-                return new $class($callParameters[0], $callParameters[1], $callParameters[2]);
-            default:
-                return $definition->getReflection()->newInstanceArgs($callParameters);
-        }
+        return new $class(...$callParameters);
```

Comment:
Since PHP 5.6 we can unpack arguments from an array with the splat operator. This improves readability, maintainability and avoids reflection usage